### PR TITLE
Expand Gluster Cluster(Tendrl-issue-id: #471)

### PR DESCRIPTION
**Pull request by gnehapk**
Tendrl-issue-id: #471 

Solved Bugs - 
tendrl-bug-id : #469, #462 , #463 , #472,  #474

### CODE REVIEW

File: src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+                interfaces = Object.keys(host.networks),
+                len = interfaces.length;
+
+            for (i = 0; i < len; i++) {
+                host.networks[interfaces[i]].ipv4 = JSON.parse(host.networks[interfaces[i]].ipv4);
+            }
+        }
+
+        /** Create host list after manipulating the API response **/
+        function _createHostList(host) {
```
Comment (cloudbehl) [inline src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js]: To large function, needs to be divided into smaller functions.

https://softwareengineering.stackexchange.com/questions/133404/what-is-the-ideal-length-of-a-method-for-you

File: src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        }
+
+        /** Returns available capacity **/
+        function _getACapacity(host) {
+            var keys = Object.keys(host.localstorage.blockdevices.free),
+                len = keys.length,
+                disks = host.localstorage.blockdevices.free,
+                size = 0,
+                i,
+                disk;
```
Comment (cloudbehl) [inline src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js]: I think "disk" is a unused variable.

File: src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+        /** Returns free disks **/
+        function _getDisks(host) {
+            var keys = Object.keys(host.localstorage.blockdevices.free),
+                len = keys.length,
+                disks = host.localstorage.blockdevices.free,
+                temp,
+                conf = [],
+                i,
+                disk;
```
Comment (cloudbehl) [inline src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js]: I think "disk" is a unused variable.

File: src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js (hunk 1 above)
Comment (gnehapk) [inline src/modules/cluster/expand-gluster-cluster/expand-gluster-cluster.js]: Done

**Pull request merged**